src/utility/CurationUtility.js

```diff
@@ -0,0 +1,231 @@
+
+  // convert mechanism evidence from object to array of objects and include evidence that have non empty description or non empty evidence types
+  let mechanismEvidenceArray = [];
+  for (const [publicationPmid, valueObj] of Object.entries(
+    preparedInput.mechanism_evidence
+  )) {
+    let evidenceTypesArray = [];
+    for (const [primaryType, secondaryTypesArray] of Object.entries(
+      valueObj.evidence_types
+    )) {
+      if (secondaryTypesArray.length > 0) {
+        let evidenceTypeObj = {
+          primary_type: primaryType,
+          secondary_type: secondaryTypesArray,
+        };
+        evidenceTypesArray.push(evidenceTypeObj);
+      }
+    }
+    if (valueObj.description !== "" || evidenceTypesArray.length > 0) {
+      let mechanismEvidenceObj = {
+        pmid: publicationPmid,
+        description: valueObj.description,
+        evidence_types: evidenceTypesArray,
+      };
+      mechanismEvidenceArray.push(mechanismEvidenceObj);
+    }
+  }
+  preparedInput.mechanism_evidence = mechanismEvidenceArray;
+
+  // convert variant types from object to array of objects and include variant types that have any non empty field data
+  let variantTypesArray = [];
+  for (const [primaryType, primaryTypeValue] of Object.entries(
+    preparedInput.variant_types
+  )) {
+    for (const [secondaryType, secondaryTypeValue] of Object.entries(
+      primaryTypeValue
+    )) {
+      if (
+        secondaryTypeValue.de_novo ||
+        secondaryTypeValue.inherited ||
+        secondaryTypeValue.nmd_escape ||
+        secondaryTypeValue.unknown_inheritance ||
+        secondaryTypeValue.comment !== "" ||
+        secondaryTypeValue.supporting_papers.length > 0
+      ) {
+        let variantTypesObj = { ...secondaryTypeValue };
+        variantTypesObj.primary_type = primaryType;
+        variantTypesObj.secondary_type = secondaryType;
+        variantTypesArray.push(variantTypesObj);
+      }
+    }
+  }
+  preparedInput.variant_types = variantTypesArray;
+
+  // convert variant descriptions from object to array of objects and include variant descriptions that have non empty description
+  preparedInput.variant_descriptions = convertObjectWithPmidToArrayOfObjects(
+    preparedInput.variant_descriptions
+  ).filter((item) => item.description !== "");
+
+  // convert variant consequences from object to array of objects and include variant consequence that have non empty support
+  let variantConsequencesArray = [];
+  for (const [key, value] of Object.entries(
+    preparedInput.variant_consequences
+  )) {
+    if (value !== "") {
+      let variantConsequencesObj = {
+        name: key,
+        support: value,
+      };
+      variantConsequencesArray.push(variantConsequencesObj);
+    }
+  }
+  preparedInput.variant_consequences = variantConsequencesArray;
+
+  // prefix locus to disease name
+  preparedInput.disease.disease_name = `${preparedInput.locus.toUpperCase()}-related ${
+    preparedInput.disease.disease_name
```

Comment:
If we don't input a disease name, it sends the disease name = `<gene>-related`.

In case the disease is not inserted, it should send:
`"disease": {}` or `"disease": null`

Reply:
taken care in latest commit